test/new-e2e/utils/payloadtesthelper/checkRunPayloads.go

```diff
@@ -0,0 +1,74 @@
+package payloadtesthelper
```

Comment:
Agree, but was planning on improving and generalising in a follow up PR. I tend to go with a couple of similar implementation and generalise with the third as a rule of thumb to prevent myself from over engineering things. Ok for handling this in a follow up ?

Reply:
As most of the code in this PR is duplicated and using generic in this context is not complicated, I think it would be great to make the changes in this PR. Also, another user seeing this pattern will tend to copy past the code.